alpha/lib/model/Partner.php

```diff
@@ -129,7 +129,13 @@ public function validateSecret ( $partner_secret , $partner_key , &$ks_max_expir
 			in_array($partner_secret, $additionalSecrets, true) ||
 			(!$admin && $partner_secret === $this->getSecret()))
 		{
-			$ks_max_expiry_in_seconds = $this->getKsMaxExpiryInSeconds();
+			$ks_max_expiry_in_seconds_from_partner = $this->getKsMaxExpiryInSeconds();
+			if (!$ks_max_expiry_in_seconds_from_partner) {
```

Comment:
i think that in that case we need to keep the old behavior and just return the $this->getKsMaxExpiryInSeconds() as it was. we only want to add an upper bound to the expiry in case it was set